pulp_container/app/registry_api.py

```diff
@@ -803,93 +793,157 @@ def put(self, request, path, pk=None):
-            models.MEDIA_TYPE.CONFIG_BLOB,
-            models.MEDIA_TYPE.CONFIG_BLOB_OCI,
+
+        if request.content_type in (
+            models.MEDIA_TYPE.MANIFEST_LIST,
+            models.MEDIA_TYPE.INDEX_OCI,
         ):
-            raise BlobInvalid(digest=config_blob.digest)
+            manifest_list = models.Manifest(
+                digest=manifest_digest,
+                schema_version=content_data["schemaVersion"],
+                media_type=content_data.get("mediaType", models.MEDIA_TYPE.INDEX_OCI),
+            )
+            try:
+                manifest_list.save()
+            except IntegrityError:
+                manifest_list = models.Manifest.objects.get(digest=manifest_list.digest)
+                manifest_list.touch()
 
-        manifest = models.Manifest(
-            digest=manifest_digest,
-            schema_version=2,
-            media_type=request.content_type,
-            config_blob=config_blob,
-        )
-        try:
-            manifest.save()
-        except IntegrityError:
-            manifest = models.Manifest.objects.get(digest=manifest.digest)
-            manifest.touch()
-        ca = ContentArtifact(artifact=artifact, content=manifest, relative_path=manifest.digest)
-        try:
-            ca.save()
-        except IntegrityError:
-            ca = ContentArtifact.objects.get(content=manifest, relative_path=manifest.digest)
+            ca = ContentArtifact(
+                artifact=artifact, content=manifest_list, relative_path=manifest_list.digest
+            )
+            try:
+                ca.save()
+            except IntegrityError:
+                ca = ContentArtifact.objects.get(
+                    content=manifest_list, relative_path=manifest_list.digest
+                )
+                if not ca.artifact:
+                    ca.artifact = artifact
+                    ca.save(update_fields=["artifact"])
+
+            manifests_to_list = []
+            for manifest_metadata in content_data.get("manifests"):
+                manifest = models.Manifest.objects.get(digest=manifest_metadata["digest"])
+
+                platform = manifest_metadata["platform"]
+                manifest_to_list = models.ManifestListManifest(
+                    manifest_list=manifest,
+                    image_manifest=manifest_list,
+                    architecture=platform["architecture"],
+                    os=platform["os"],
+                    features=platform.get("features", ""),
+                    variant=platform.get("variant", ""),
+                    os_version=platform.get("os.version", ""),
+                    os_features=platform.get("os.features", ""),
+                )
+                manifests_to_list.append(manifest_to_list)
+
+            models.ManifestListManifest.objects.bulk_create(
+                manifests_to_list, ignore_conflicts=True, batch_size=1000
+            )
+
+            tag = models.Tag(name=pk, tagged_manifest=manifest_list)
```

Comment:
if you rename the variable on line 805 from manifest_list to manifest you can use L850-L871 without repeating it twice. just move this code at the bottom